R/plotLoadings.R

```diff
@@ -0,0 +1,362 @@
+                        n = n,
+                        ncomponents = ncomponents,
+                        ...)
+                      
+        # Checking if there are enough components in the matrix
+        .check_components(x, ncomponents)
+        # Ordering loadings and adding factor to keep the order
+        df <- .get_loadings_plot_data(x, n, ncomponents)
+        # Plot features with the layout selected
+        p <- .plot_pca_feature_loadings(df, layout, n, ncomponents)
+        
+        return(p)
+    }
+)
+
+.check_parameters <- function(x, dimred, layout, n, ncomponents, tree.name,...) {
+    # Check tree.name
+    if( is(x, "TreeSummarizedExperiment") && !(tree.name %in% rowTreeNames(x)  && .is_a_string(tree.name))){
+        stop("'tree.name' must be a single character value specifying a colTree.", call. = FALSE)
+    }
+    # Checking if dimred is correct
+    if( is(x, "TreeSummarizedExperiment") && !(dimred %in% reducedDimNames(x)  && .is_a_string(dimred))){
+        stop("'dimred' must specify reducedDim.", call. = FALSE)
+    }
+    # Checking if layout is correct
+    if ( !(layout %in% c("screeplot", "barplot", "tree", "heatmap") && .is_a_string(layout)) ) {
+        stop("'layout' must be one of c('screeplot', 'barplot', 'tree', 'heatmap').", call. = FALSE)
+    }
+    # Making sure the user doesn't try to plot the tree if he gives only the matrix
+    if (is.matrix(x) && layout == "tree") {
+        stop("TreeSummarizedExperiment object is required for the tree plotting.", call. = FALSE)
+    }
+    # Making sure the tree is not null
+    if( layout == "tree" && !(is(x, "TreeSummarizedExperiment") && !is.null(rowTree(x, tree.name)) )) {
+        stop ("Tree is null.", call. = FALSE)
+    }
+    # Checking if n is a positive number
+    if ( !(is.numeric(n) && n > 0) ) {
+        stop("'n' must be a positive number.", call. = FALSE)
+    }
+    # Checking if ncomponents is a positive number 
+    if ( !(is.numeric(ncomponents) && ncomponents > 0) ) {
+        stop("'ncomponents' must be a positive number.", call. = FALSE)
+    }
+    return(NULL)
+
+}
+
+.check_components <- function(x, ncomponents) {
+    # Checking if there are enough components in the matrix
+    if (ncomponents > ncol(x)) {
+        stop("'ncomponents' must be lower or equal than number of components.", call. = FALSE)  
+    }
+    return(NULL)
+}
+
+# Function to process each component
+.process_component <- function(i, df, n) {
+    # Ordering loadings by absolute value
+    df <- df[order(-abs(df[[i]])), ][1:n, ]
+    # Ordering by actual loadings
+    df <- df[order(df[[i]]), ]
+    # Add factor to keep order
+    df[["Feature"]] <- factor(df[["Feature"]], levels = df[["Feature"]])
+    return(df)
+}
+
+.get_loadings_plot_data <- function(x, n, ncomponents) {
+  # Transform into a dataframe
+  x <- as.data.frame(x)
+  # Keep only the number of components needed
+  names <- colnames(x)[1:ncomponents]
+  df <- dplyr::select(x, all_of(names))
```

Comment:
Instead of dplyr::select(), use importFrom above the function

@importFrom dplyr select